gtsfm/frontend/matcher/loftr.py

```diff
@@ -71,8 +72,15 @@ def match(self, image_i1: Image, image_i2: Image) -> Tuple[Keypoints, Keypoints]
 
         keypoints_i1 = Keypoints(coordinates=coords_i1_by_confidence)
         keypoints_i2 = Keypoints(coordinates=coords_i2_by_confidence)
-
-        return keypoints_i1, keypoints_i2
+        valid_ind = np.arange(len(keypoints_i1))
+        if image_i1.mask is not None:
+            _, valid_ind_i1 = keypoints_i1.filter_by_mask(image_i1.mask)
+            valid_ind = np.intersect1d(valid_ind, valid_ind_i1)
+        if image_i2.mask is not None:
+            _, valid_ind_i2 = keypoints_i2.filter_by_mask(image_i2.mask)
+            valid_ind = np.intersect1d(valid_ind, valid_ind_i2)
```

Comment:
thanks for the updates @travisdriver. i wonder if we should put the masking at a higher level, like in the CorrespondenceGenerator, after all detections are generated? I think we only handle it in SuperPoint elsewhere for now

Reply:
I agree; should I do it in this PR or a separate one?

Reply:
Separate one is fine with me, can you open an issue for it too? Thanks!